corehq/apps/case_search/xpath_functions/value_functions.py

```diff
@@ -115,3 +166,30 @@ def unwrap_list(node, context):
 
     value = node.args[0]
     return json.loads(value)
+
+
+def double(node, context):
+    assert node.name == 'double'
+    confirm_args_count(node, 1)
+    value = unwrap_value(node.args[0], context)
+
+    if isinstance(value, str):
+        try:
+            parsed_date = parse_date(value)
+        except ValueError:
+            ...
```

Comment:
interesting. I this is fine because `parsed_date` will be None if an exception is thrown?

Reply:
it's kind of annoying actually - `parse_date` can return either `None` _or_ throw an exception.  Ah so you're right - that means this will fail if it throws an exception huh?  dang, let me fix that